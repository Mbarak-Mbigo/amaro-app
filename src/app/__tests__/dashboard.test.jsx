import React from "react";
import { Provider } from "react-redux";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import configureMockStore from "redux-mock-store";

import Dashboard from "../dashboard";

const mockStore = configureMockStore([]);

describe("DashBoard Component", () => {
  const store = mockStore({
    cart: {
      selected: []
    },
    products: {
      data: [],
      isLoading: false,
      hasLoaded: false,
      error: {
        hasError: false,
        error: ''
      }
    }
  });
  test("renders Dashboard component", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Dashboard />
        </Router>
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("Renders 2 children", () => {
    const wrapper = shallow(<Dashboard />);
    expect(wrapper.children()).toHaveLength(2);
  });
});
