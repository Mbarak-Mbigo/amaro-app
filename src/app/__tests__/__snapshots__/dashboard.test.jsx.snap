// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashBoard Component renders Dashboard component 1`] = `
<div>
  <div
    className="sc-gqjmRU jlTudv"
  >
    <div
      className="sc-VigVT ieyXlk"
    >
      <a
        className="sc-iwsKbI hLjuIW"
        href="/cart"
        onClick={[Function]}
      >
        View Cart
      </a>
      <a
        className="sc-iwsKbI hLjuIW"
        href="/cart"
        onClick={[Function]}
      >
        Selected Items: 0
      </a>
    </div>
    <input
      className="sc-jTzLTM eewZEe sc-gZMcBi hpIRxJ"
      name="searchBox"
      onChange={[Function]}
      placeholder="Search products"
    />
  </div>
</div>
`;
